src/sentry_plugins/sessionstack/static/sessionstack/contexts/sessionstack.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+const ASPECT_RATIO = 16 / 9;
+
+const SessionStackContextType = React.createClass({
+  propTypes: {
+    alias: React.PropTypes.string.isRequired,
+    data: React.PropTypes.object.isRequired
+  },
+
+  getInitialState() {
+    return {
+      showIframe: false
+    };
+  },
+
+  componentDidMount() {
+    this.parentNode = ReactDOM.findDOMNode(this).parentNode;
+    window.addEventListener('resize', this.setIframeSize, false);
+    this.setIframeSize();
+  },
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.setIframeSize, false);
+  },
+
+  setIframeSize() {
+    if (!this.showIframe) {
+      let parentWidth = $(this.parentNode).width();
+
+      this.setState({
+        width: parentWidth,
+        height: parentWidth / ASPECT_RATIO
+      });
+    }
+  },
+
+  playSession() {
+    this.setState({
+      showIframe: true
+    });
+
+    this.setIframeSize();
+  },
+
+  render() {
+    let { session_url } = this.props.data;
+
+    if (!session_url) {
+      return (
+        <h4>Session not found.</h4>
+      );
+    }
+
+    return (
+      <div className="panel-group">
+        {this.state.showIframe ?
+          <iframe src={session_url} width={this.state.width} height={this.state.height} /> :
```

Comment:
So without any other investigation, this is going to be very problematic for us.

On sentry.io specifically, we have a very strict CSP rule to prevent injecting random content into our pages, whether from XSS or anything else to compromise security. This is taking an arbitrary url as input from the user, and injecting it as an iframe. I'm not entirely sure of the consequences of someone using `https://maliciousdomain.com` as the player_url here. I think it'd have access to the `parent` frame and be able to execute JavaScript. I could be wrong here, but our CSP rules would reject it anyways.

Second, the default value of this is an http url, which will be blocked regardless of CSP rule since we are served over https only.

Given these constraints, is there a way that we can make this work? Can we proxy through our servers or something, or render an iframe via Sentry somehow.

/cc @benvinegar since he knows much more about iframe security than I do.